### ISSUE TITLE
TS error if .proto file has the same message name as imported 

### ISSUE THREAD

**Issue by skycrazyk** [stephenh/ts-proto#896]
### Dependencies 

- ts-proto@1.156.2
- typescript@5.1.6
- protobufjs@7.2.4

### Problem description

There is `common/common.proto` which has `FindOptions` message

```proto
syntax = "proto3";
package common;

message FindOptions {
  repeated string sort = 1;
  int32 page_num = 2;
  int32 page_size = 3;
}
```

and `items/items.proto` which has `FindOptions` too. Also `items.proto` imports `common.proto` and uses `common.FindOptions` message inside their own `FindOptions` message.

```proto
syntax = "proto3";
import "common/common.proto";

package content.items;

message FindOptions {
  common.FindOptions options = 2;
  bool regular = 4;
  bool hidden = 5;
  bool templates = 6;
}

message FindRequest {
  string space_id = 1;
  FindOptions options = 5;
}
```

Finally during the build process we have TS error

<details>
  <summary>(spoiler) config/dist/items/items.ts(1511,27): error TS2345: Argument of type 'import("/builds/perxis/perxis- 
  js/config/dist/items/items").FindOptions' is not assignable to parameter of type 'import("/builds/perxis/perxis- 
  js/config/dist/common/common").FindOptions'.</summary>
  
  ```
  config/dist/items/items.ts(1511,27): error TS2345: Argument of type 'import("/builds/perxis/perxis- 
  js/config/dist/items/items").FindOptions' is not assignable to parameter of type 'import("/builds/perxis/perxis- 
  js/config/dist/common/common").FindOptions'.
  Type 'FindOptions' is missing the following properties from type 'FindOptions': sort, pageNum, pageSize, fields, excludeFields
  config/dist/items/items.ts(1540,11): error TS2739: Type 'FindOptions' is missing the following properties from type 
  'FindOptions': deleted, regular, hidden, templates
config/dist/items/items.ts(1584,5): error TS2322: Type 'import("/builds/perxis/perxis-js/config/dist/common/common").FindOptions | undefined' is not assignable to type 'import("/builds/perxis/perxis-js/config/dist/items/items").FindOptions | undefined'.
  Type 'import("/builds/perxis/perxis-js/config/dist/common/common").FindOptions' is not assignable to type 'import("/builds/perxis/perxis-js/config/dist/items/items").FindOptions'.
config/dist/items/items.ts(1585,34): error TS2559: Type '{ options?: ... | undefined; deleted?: boolean | undefined; regular?: boolean | undefined; hidden?: boolean | undefined; templates?: boolean | undefined; }' has no properties in common with type '{ sort?: string[] | undefined; pageNum?: number | undefined; pageSize?: number | undefined; fields?: string[] | undefined; excludeFields?: boolean | undefined; }'.
config/dist/items/items.ts(1883,27): error TS2345: Argument of type 'import("/builds/perxis/perxis-js/config/dist/items/items").FindOptions' is not assignable to parameter of type 'import("/builds/perxis/perxis-js/config/dist/common/common").FindOptions'.
config/dist/items/items.ts(1912,11): error TS2322: Type 'import("/builds/perxis/perxis-js/config/dist/common/common").FindOptions' is not assignable to type 'import("/builds/perxis/perxis-js/config/dist/items/items").FindOptions'.
config/dist/items/items.ts(1956,5): error TS2322: Type 'import("/builds/perxis/perxis-js/config/dist/common/common").FindOptions | undefined' is not assignable to type 'import("/builds/perxis/perxis-js/config/dist/items/items").FindOptions | undefined'.
config/dist/items/items.ts(1957,34): error TS2559: Type '{ options?: ... | undefined; deleted?: boolean | undefined; regular?: boolean | undefined; hidden?: boolean | undefined; templates?: boolean | undefined; }' has no properties in common with type '{ sort?: string[] | undefined; pageNum?: number | undefined; pageSize?: number | undefined; fields?: string[] | undefined; excludeFields?: boolean | undefined; }'.
config/dist/items/items.ts([197](https://git.perx.ru/perxis/perxis-js/-/jobs/105621#L197)4,27): error TS2345: Argument of type 'import("/builds/perxis/perxis-js/config/dist/items/items").FindOptions' is not assignable to parameter of type 'import("/builds/perxis/perxis-js/config/dist/common/common").FindOptions'.
config/dist/items/items.ts([199](https://git.perx.ru/perxis/perxis-js/-/jobs/105621#L199)1,11): error TS2322: Type 'import("/builds/perxis/perxis-js/config/dist/common/common").FindOptions' is not assignable to type 'import("/builds/perxis/perxis-js/config/dist/items/items").FindOptions'.
config/dist/items/items.ts([200](https://git.perx.ru/perxis/perxis-js/-/jobs/105621#L200)7,5): error TS2322: Type 'import("/builds/perxis/perxis-js/config/dist/common/common").FindOptions | undefined' is not assignable to type 'import("/builds/perxis/perxis-js/config/dist/items/items").FindOptions | undefined'.
config/dist/items/items.ts(2008,34): error TS2559: Type '{ options?: ... | undefined; deleted?: boolean | undefined; regular?: boolean | undefined; hidden?: boolean | undefined; templates?: boolean | undefined; }' has no properties in common with type '{ sort?: string[] | undefined; pageNum?: number | undefined; pageSize?: number | undefined; fields?: string[] | undefined; excludeFields?: boolean | undefined; }'.
config/dist/items/items.ts([202](https://git.perx.ru/perxis/perxis-js/-/jobs/105621#L202)1,27): error TS2345: Argument of type 'import("/builds/perxis/perxis-js/config/dist/items/items").FindOptions' is not assignable to parameter of type 'import("/builds/perxis/perxis-js/config/dist/common/common").FindOptions'.
config/dist/items/items.ts([203](https://git.perx.ru/perxis/perxis-js/-/jobs/105621#L203)8,11): error TS2322: Type 'import("/builds/perxis/perxis-js/config/dist/common/common").FindOptions' is not assignable to type 'import("/builds/perxis/perxis-js/config/dist/items/items").FindOptions'.
config/dist/items/items.ts([205](https://git.perx.ru/perxis/perxis-js/-/jobs/105621#L205)4,5): error TS2322: Type 'import("/builds/perxis/perxis-js/config/dist/common/common").FindOptions | undefined' is not assignable to type 'import("/builds/perxis/perxis-js/config/dist/items/items").FindOptions | undefined'.
config/dist/items/items.ts(2055,34): error TS2559: Type '{ options?: ... | undefined; deleted?: boolean | undefined; regular?: boolean | undefined; hidden?: boolean | undefined; templates?: boolean | undefined; }' has no properties in common with type '{ sort?: string[] | undefined; pageNum?: number | undefined; pageSize?: number | undefined; fields?: string[] | undefined; excludeFields?: boolean | undefined; }'.
```
</details>

The problem that final `items.d.ts` file doesn't imports `FindOptions` from `common` at all

![image](https://github.com/stephenh/ts-proto/assets/7424691/311a0aea-311f-475b-92d7-84789d539b63)

so inside `items.d.ts` `FindOptions` interface refers on themself 

![image](https://github.com/stephenh/ts-proto/assets/7424691/837da0ec-634e-4eba-ab5a-462d02a30b7e)

I have similar story with `Filter` message, which also exists in both files, but inside `items.proto` it is used with `repeated`, and in final `items.d.ts` it imported correctly, with rename to `Filter1` as you can see on the image above.

**Comment by stephenh** [comment]
Huh, yeah, we're supposed to have the infra in place to support this...the ts-poet codegen library knows about "symbols defined in this file" as well as "symbols imported from other files" and will rename the imports if it detects a conflict.

I.e. that is why `Filter` is getting renamed to `Filter1` on import, is that ts-poet realized there is already a `Filter` name in-scope, so it synthesized a different/unique name for the import, and then in theory in code that would have using `Filter`-from-`common` will use `Filter1` instead.

It seems like that did not work for `FindOption`, like it seems to not realize that "there are two `FindOption`s" in the first place.

The code that seems `items.FindOption` has a field of `common.FindOption` is here:

https://github.com/stephenh/ts-proto/blob/main/src/main.ts#L875

Which eventually gets here:

https://github.com/stephenh/ts-proto/blob/main/src/types.ts#L586

And should be getting `.common.FindOptions` as the `protoType` of that field, which should know it comes from another module, and so should generate an `impProto` that ends up creating an `import` statement in the output.

If you'd like to track down where that is going wrong, that would be great! Thanks!

**Comment by skycrazyk** [comment]
I find out that everything worked fine until version 1.152.0. 

In version 1.152.0 there is a bug, and I have error during build process

![image](https://github.com/stephenh/ts-proto/assets/7424691/c5e78a32-0abc-49dc-932a-69dbc52c14f4)

and then in version 1.152.1 I see problem described in this issue

![image](https://github.com/stephenh/ts-proto/assets/7424691/d0f5d955-1ae6-464e-acb3-ad7f6531aa8a)

**Comment by xmkhan** [comment]
+1 Noticing the same problem. It now fails to compile because the toJSON and fromJSON fail to use the alias'ed import as the type.

**Comment by andrehp** [comment]
I'm having the same issue, it's easy to add a case to fail the tests, you just need to add this message to `integration/avoid-import-conflicts/simple.proto`:

```
message DifferentSimple {
  string name = 1;
  optional simple2.Simple otherSimple = 2;
}
```

The generated interface is incorrect:
```
export interface DifferentSimple {
  name: string;
  otherSimple?: Simple | undefined;
}
```

otherSimple should be Simple3 | undefined instead of Simple | undefined

**Comment by andrehp** [comment]
I've opened the PR to fix this issue [here](https://github.com/stephenh/ts-proto/pull/904). It seems that the issue was the check to emit a warning, which transforms the Code object.

I've also tested it with the example provided by skycrazyk and it generates the code correctly now.

**Comment by stephenh** [comment]
I just merged @andrehp 's fix, #904 , so report back if the latest release doesn't fix your issue. Thanks @andrehp !
